src/pudl/etl/glue_assets.py

```diff
@@ -170,6 +177,7 @@ def epacamd_eia(
         .pipe(pudl.metadata.fields.apply_pudl_dtypes, "eia")
         .dropna(subset=["plant_id_eia"])
         .pipe(correct_epa_eia_plant_id_mapping)
+        .drop_duplicates()
```

Comment:
It's certainly possible, and I even just did a little bit of digging and found some records where something like this is happening, with an example below:

| plant_id_epa | emission_unit_id_epa | generator_id_epa | plant_id_eia | boiler_id | generator_id |     year     |
|         ---            |                    ---                 |               ---               |          ---          |      ---       |            ---          |      ---        |
|         63628     |                    5A-1            |               5A              |         2953      |                 |            5A         |     2018    |
|         63628     |                    5A-1            |               5A              |         63628    |                 |            5A         |     2021    | 

Given that genuine 1:m matches are common and expected, these feel very difficult to detect. I'm also not sure that we'd want to get rid of them. In a case like the one above, it seems like it could be genuinely useful to keep both of these matches as they might reflect changes in the underlying data. Perhaps adding a year column could help differentiate between actual 1:m matches, and slightly different matches from different years.

Reply:
hmmm seems very weird that a generator/emissions unit id would map to a completely different plant. Can you dig into what plant that is? 

Generally speaking, I think I'm in favor of adding the year column.

Reply:
Here are the two plants in question:

plant_id_eia | plant_name_eia | plant_id_pudl
-- | -- | --
2953 | Mustang | 403
63628 | tinker | 11586

If we look at the generators for each of these plants:

plant_id_eia | plant_name_eia | generator_id
-- | -- | --
2953 | Mustang | 1
2953 | Mustang | 2
2953 | Mustang | 3
2953 | Mustang | 4
2953 | Mustang | GT1
2953 | Mustang | GT2
2953 | Mustang | GT3
2953 | Mustang | GT4
2953 | Mustang | GT5
2953 | Mustang | GT6
2953 | Mustang | GT7
2953 | Mustang | SLR1
63628 | tinker | 5A
63628 | tinker | 5B

It seems that the old match (63628->2953) is actually incorrect, given that it doesn't have a generator with ID `5A`, but this is what's in the version of the crosswalk with just 2018 data. I'm not sure exactly how you would detect/correct cases like this in an automated way. There aren't too many cases that seem to be obvious candidates for this kind of mismatch (<30 I think), so maybe we could just investigate each of them individually and add any known errors to the manual matches, which we could even try to contribute back to the EPA repo.

> Can you update the description of this PR to explain why it was bad to have dropped connections from the 2018 data

If I remember correctly, I found over 100 cases where there were matches in 2018, but no matches in 2021, and that doesn't seem optimal to me.

In general, I think there might need to be a bigger discussion about how we want to handle multiple years though, so we might want to just stick with one year to get the builds passing again, then decide how to go forward.

Reply:
If we aspire to having better multi-year coverage I think adding a `report_year` column is a great idea. We've found unexpected instances where associations do slowly change over time, sometimes for good reasons. It'll be easy to simplify that information to the version with no years and no duplicates if we want to. Storing the per-year data in the DB table seems like it gives us the most downstream flexibility.

Reply:
Yeah, for now adding a year column seems like the easiest fix. But maybe then we don't want to drop duplicates?

Reply:
If it's genuinely a bad match though, we should probably have a way to fix that.